test/client/js/helpers/messageMatchFilters.js

```diff
@@ -0,0 +1,41 @@
+"use strict";
+
+const expect = require("chai").expect;
+const messageMatchFilters = require("../../../../client/js/helpers/messageMatchFilters").default;
+
+describe("messageMatchFilters helper", function () {
+	const hidePreviewFilters = "foo, bar,   ns fw   , 고";
+
+	it("should NOT match a filter", function () {
+		const teststrings = [
+			"http://url.com/nsfw.png", // Only "ns fw" was provided, the space was not removed
+			"nsfw http://url.com/image.gif",
+			"ns http://url.com/pic.png fw",
+			"http://url.com/link.png",
+			"hey check this out http://url.com/link.ogg",
+			"this is sfw http://url.com/sfw.mp4",
+			"foö http://url.com/link.mp4",
+			"win http://url.com/song.mp3 dows",
+		];
+
+		for (const teststring of teststrings) {
+			expect(messageMatchFilters(teststring, hidePreviewFilters)).to.be.false;
+		}
+	});
+
+	it("should match a filter", function () {
+		const teststrings = [
+			"this http://www.bar.com/link.png should be filtered",
```

Comment:
I'm wondering if that's really desired.
A word filter should match on words, not arbitrary substrings.

Meaning "this bar should be filtered" while "this foobar should not"